File: programming_assignment_1/fibonacci.rb
def calc_fib(n)
  if n == 0 return 0

  fibs = [0, 1]
  while fibs.length <= n do
    next_fib = fibs[-1] + fibs[-2]
    fibs << next_fib
  end
  fibs.last
end

if __FILE__ == $0
  n = gets.to_i
  puts "#{calc_fib(n)}"
end

File: programming_assignment_1/fibonacci.py
# Uses python3
def calc_fib(n):
    fibs = [0, 1]
    while len(fibs) <= n:
        next_fib = fibs[-1] + fibs[-2]
        fibs.append(next_fib)
    return fibs[-1]

n = int(input())
print(calc_fib(n))
